lib/client/api.go

```diff
@@ -2694,43 +2727,119 @@ func (tc *TeleportClient) runCommandOnNodes(ctx context.Context, clt *ClusterCli
 		return trace.Wrap(err)
 	}
 
+	if tc.SSHLogDir != "" {
+		if err := os.MkdirAll(tc.SSHLogDir, 0755); err != nil {
+			return trace.ConvertSystemError(err)
+		}
+	}
+
+	resultsCh := make(chan execResult, len(nodes))
+
 	g, gctx := errgroup.WithContext(ctx)
 	g.SetLimit(commandLimit(ctx, clt.AuthClient, mfaRequiredCheck.Required))
-	for _, address := range nodeAddresses {
-		address := address
+	for _, node := range nodes {
+		node := node
 		g.Go(func() error {
 			ctx, span := tc.Tracer.Start(
 				gctx,
 				"teleportClient/executingCommand",
 				oteltrace.WithSpanKind(oteltrace.SpanKindClient),
-				oteltrace.WithAttributes(attribute.String("node", address)),
+				oteltrace.WithAttributes(attribute.String("node", node.addr)),
 			)
 			defer span.End()
 
 			nodeClient, err := tc.ConnectToNode(
 				ctx,
 				clt,
 				NodeDetails{
-					Addr:      address,
+					Addr:      node.addr,
 					Namespace: tc.Namespace,
 					Cluster:   cluster,
 					MFACheck:  mfaRequiredCheck,
+					hostname:  node.hostname,
 				},
 				tc.Config.HostLogin,
 			)
 			if err != nil {
 				fmt.Fprintln(tc.Stderr, err)
-				return trace.Wrap(err)
+				return nil
 			}
 			defer nodeClient.Close()
 
-			fmt.Printf("Running command on %v:\n", nodeName(address))
+			displayName := nodeName(node.addr)
+			if node.hostname != "" {
+				displayName = node.hostname
+			}
+			fmt.Printf("Running command on %v:\n", displayName)
 
-			return trace.Wrap(nodeClient.RunCommand(ctx, command))
+			if err := nodeClient.RunCommand(ctx, command); err != nil && tc.ExitStatus == 0 {
+				fmt.Fprintln(tc.Stderr, err)
+				return nil
```

Comment:
Same question as above - this seems like a behavior change. Is it intentional? If so, why?